### PR TITLE
Implements W.from_WSP

### PR DESCRIPTION
**Pull request by sjsrey**
Addresses #495

### CODE REVIEW

File: libpysal/weights/weights.py [hunk 1]
```diff
@@ -232,7 +232,69 @@ def from_shapefile(cls, *args, **kwargs):
...
+
+        Create a standard PySAL W from this sparse weights object.
+
+        >>> w = W.from_WSP(wsp)
+        >>> w.n
+        10
+        >>> print(w.full()[0][0])
+        [0 1 0 0 0 1 0 0 0 0]
+        """
+        WSP.sparse
```
Comment (jGaboardi) [inline libpysal/weights/weights.py:275]: Is there a reason for calling `WSP.sparse` here?

File: libpysal/weights/weights.py [hunk 2]
```diff
@@ -232,7 +232,69 @@ def from_shapefile(cls, *args, **kwargs):
...
+        indices = WSP.sparse.indices
```
Comment (jGaboardi) [inline libpysal/weights/weights.py:276]: Seems like `indices` does not need to be assigned here and `WSP.sparse.indices` can be used directly in line 282:

```python 
indices = [id_order[i] for i in WSP.sparse.indices]
```

File: libpysal/weights/weights.py (hunk 1 above)
Comment (sjsrey) [reply libpysal/weights/weights.py:275]: good catch. I was pulling this over from https://github.com/pysal/libpysal/blob/master/libpysal/weights/util.py#L788 .

Will update here and there.

File: libpysal/weights/weights.py (hunk 2 above)
Comment (sjsrey) [reply libpysal/weights/weights.py]: Good catch. Will change.

**Review by jGaboardi — APPROVED** [review]

**Comment by jGaboardi** [comment]
Maybe pulled the approval trigger too fast. We should figure a work around for failures in 3.7.

**Comment by sjsrey** [comment]
> Maybe pulled the approval trigger too fast. We should figure a work around for failures in 3.7.

Or we could consider dropping support for 3.7?

**Comment by jGaboardi** [comment]
> > Maybe pulled the approval trigger too fast. We should figure a work around for failures in 3.7.
> 
> 
> 
> Or we could consider dropping support for 3.7?

I 100% support this.

**Comment by knaaptime** [comment]
cool with me. its outside the 'last three' release window we commit to

**Pull request merged**